application/src/main/java/org/opentripplanner/gtfs/mapping/TimeframeMapper.java

```diff
@@ -0,0 +1,43 @@
+package org.opentripplanner.gtfs.mapping;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+import java.util.Collection;
+import org.opentripplanner.core.model.id.FeedScopedId;
+import org.opentripplanner.ext.fares.model.Timeframe;
+
+class TimeframeMapper {
+
+  private final IdFactory idFactory;
+  private final Multimap<FeedScopedId, Timeframe> mappedTimeframes = ArrayListMultimap.create();
+
+  public TimeframeMapper(IdFactory idFactory) {
+    this.idFactory = idFactory;
+  }
+
+  public Timeframe map(org.onebusaway.gtfs.model.Timeframe rhs) {
+    final FeedScopedId serviceId = idFactory.createId(
+      rhs.getServiceId(),
+      "fare leg rule's timeframe"
+    );
+    var t = Timeframe.of()
+      .withServiceId(serviceId)
+      .withStart(rhs.getStartTime())
+      .withEnd(rhs.getEndTime())
```

Comment:
Not sure if that's an issue with OBA, a blank `end_time` value will map to midnight "00:00:00" (same as blank `start_time` values), so this will fail timeframe matching where `end_time` is omitted.

Reply:
That's a good catch. I had set the default to LocalDate.MIDNIGHT and had assumed that that means 24:00 but it is the same as LocalDate.MIN.

I had seconds thoughts about the OBA default values and have removed them: https://github.com/OneBusAway/onebusaway-gtfs-modules/pull/431

The handling of the missing value now happens in the OTP mapping process: bf0d9a9ec9927185e75d129ec91840446ec1efc7